### PR TITLE
Clean stored Balsamic analyses

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR adds cleaning of stored Balsamic analyses.

**How to prepare for test**:
- [x] ssh to hasta (depending on type of change)
- [x] install on stage:
`bash servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

**How to test cleaning of one case**:
- [x] login to hasta
- [x] run `us`
- [x] check that the case folder exists for CASE-ID
- [x] run `cg clean balsamic-run-dir CASE-ID -y`

**Expected test outcome**:
- [x] check that the case folder has been cleaned

**How to test cleaning of multiple cases**:
- [x] login to hasta
- [x] run `us`
- [x] check that the case folder exists
- [x] run `cg clean balsamic-past-run-dirs -d DATE`

**Expected test outcome**:
- [x] check that the feature wants to delete for a case that was completed before given date

- [x] Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by @barrystokman 
- [x] tests executed by BS + PG
- [x] "Merge and deploy" approved by PG
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by barrystokman** [comment]
**How to prepare for test**:
- [x] ssh to hasta (depending on type of change)
- [x] install on stage:
`bash servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

**How to test cleaning of one case**:
- [x] login to hasta
- [x] run `us`
- [x] check that the case folder exists for `wiredtroll` **BS**: assuming 'case folder' is `/home/proj/stage/cancer/cases/wiredtroll`
- [x] run `cg clean balsamic-run-dir wiredtroll -y`
Before:
image
After:
image

**Expected test outcome**:
- [ ] check that the case folder has been cleaned. @patrikgrenfeldt  The directory `wiredtroll` has been removed. Is this the expected behaviour?

**How to test cleaning of multiple cases**:
- [x] login to hasta
- [x] run `us`
- [ ] check that the case folder exists
- [  ] run `cg clean balsamic-past-run-dirs -d DATE`

**Expected test outcome**:
- [ ] check that the feature wants to delete for a case that was completed before given date @patrikgrenfeldt How do I determine when a case was completed?

- [ ] Take a screenshot and attach or copy/paste the output.

File: cg/cli/clean.py [hunk 1]
```diff
@@ -48,6 +53,48 @@ def beacon(context: click.Context, item_type, item_id):
...
+    analysis_path = balsamic.get_case_path(case_id)
+
+    if yes or click.confirm(f"Do you want to remove {analysis_path}?"):
+
+        if not analysis_path.exists():
+            LOG.warning("could not find: %s", analysis_path)
+            return FAIL
+
+        if analysis_path.is_symlink():
+            LOG.warning("will not automatically delete symlink: %s", analysis_path)
```
Comment (barrystokman) [inline cg/cli/clean.py:88]: Is it possible to clarify for the end user what this means or what they should do to follow up?

Comment (patrikgrenfeldt) [reply cg/cli/clean.py]: added a message to remove it manually

File: cg/cli/clean.py [hunk 2]
```diff
@@ -85,7 +85,8 @@ def balsamic_run_dir(context, yes, case_id, dry_run: bool = False):
             return FAIL
 
         if analysis_path.is_symlink():
-            LOG.warning("will not automatically delete symlink: %s", analysis_path)
+            LOG.warning("Will not automatically delete symlink: %s, delete if manually",
```
Comment (barrystokman) [inline cg/cli/clean.py]:
```suggestion
            LOG.warning("Will not automatically delete symlink: %s, delete it manually",
```

Comment (patrikgrenfeldt) [reply cg/cli/clean.py]: oops :) fixed

**Comment by barrystokman** [comment]
Exploratory testing:

Case directory doesn't exists: 
image
:heavy_check_mark: 

Dry run:
image
:heavy_check_mark: 

Analysis not found:
image
image
:heavy_check_mark: 

Case id does not exist:
image
:heavy_check_mark:

**Comment by barrystokman** [comment]
Sorry, I overlooked something...

Running `cg clean balsamic-past-run-dirs -d 2019-10-01` gives me the following:
image

These are all mip cases, wouldn't it be better to olny pcik up the balsamic cases?

Running `cg clean balsamic-past-run-dirs -d -y 2019-10-01` takes ages and will clog up log files too I assume.

**Comment by patrikgrenfeldt** [comment]
> Running `cg clean balsamic-past-run-dirs -d -y 2019-10-01` takes ages and will clog up log files too I assume.

I thought about implementing "Balsamic only" but forgot about it, now it's there :) Could you try that again?

**Comment by patrikgrenfeldt** [comment]
Re-test result:
image
👍

**Comment by patrikgrenfeldt** [comment]
Re-test clean one case:
image
👍

**Comment by patrikgrenfeldt** [comment]
Updated db:
image

Deployed to Hasta:
image

Deployed to Clinical-db:
image

**Pull request merged**